routes/project/location/postgres.rb

```diff
@@ -453,6 +453,40 @@ class Clover
           metrics: results
         }
       end
+
+      r.on "config" do
+        r.get true do
+          authorize("Postgres:view", pg.id)
+
+          {
+            pg_config: pg.user_config,
+            pgbouncer_config: pg.pgbouncer_user_config
+          }
+        end
+
+        r.put true do
+          authorize("Postgres:edit", pg.id)
+
+          pg_config = typecast_params.Hash("pg_config")
+          pgbouncer_config = typecast_params.Hash("pgbouncer_config")
+
+          pg_validator = PostgresConfig::Validator.new(pg.version)
+          pg_validator.validate(pg_config)
+
+          pgbouncer_validator = PostgresConfig::Validator.new("pgbouncer")
+          pgbouncer_validator.validate(pgbouncer_config)
+
+          pg.update(user_config: pg_config, pgbouncer_user_config: pgbouncer_config)
+          audit_log(pg, "update")
+
+          pg.servers.each(&:incr_configure)
```

Comment:
Need to check if this can trigger a restart

Reply:
So looking deeper: this should only trigger a restart if `pg_ctlcluster reload #{v} main` exits with a non-zero code. This happens when:
1. The server is not running
2. There is a critical internal error (missing files/permission errors etc)

So I think its safe to increment this semaphore for a config update. We can give the user heads up that they might need to hit the restart button in case the config they're changing requires it.